esphome/components/gdew0154m09/gdew0154m09.cpp

```diff
@@ -0,0 +1,274 @@
+  while (this->busy_pin_->digital_read()) {
+    uint32_t waittime = millis() - start;
+    if (waittime > timeout) {
+      ESP_LOGE(TAG, "Timeout while displaying image! was %dms", waittime);
+      return false;
+    }
+    delay(10);
+    App.feed_wdt();
+  }
+  return true;
+}
+
+int GDEW0154M09::clear_dsram_() {
+  int pixsize = this->get_buffer_length_();
+  for (int i = 0; i < 2; i++) {
+    command_(CMD_DTM1_DATA_START_TRANS);
+    for (int count = 0; count < pixsize; count++) {
+      data_(0x00);  // set every pixel to 0
+    }
+    delay(2);
+    command_(CMD_DTM2_DATA_START_TRANS2);  // data transmission start DTM2
+    for (int count = 0; count < pixsize; count++) {
+      data_(0xff);  // set every pixel to ff
+    }
+  }
+  return 0;
+}
+
+int GDEW0154M09::clear_(int mode) {
+  int pixsize = this->get_buffer_length_();
+  if (mode == CLEAR_MODE_FULL) {
+    this->command_(CMD_DTM1_DATA_START_TRANS);
+    for (int count = 0; count < pixsize; count++) {
+      this->data_(0xff);
+    }
+    delay(2);
+    this->command_(CMD_DTM2_DATA_START_TRANS2);
+    for (int count = 0; count < pixsize; count++) {
+      this->data_(0x00);
+    }
+
+    delay(2);
+    this->command_(CMD_DISPLAY_REFRESH);
+    this->wait_until_idle_();
+
+    this->command_(CMD_DTM1_DATA_START_TRANS);
+    for (int count = 0; count < pixsize; count++) {
+      this->data_(0x00);
+    }
+    delay(2);
+    this->command_(CMD_DTM2_DATA_START_TRANS2);
+    for (int count = 0; count < pixsize; count++) {
+      this->data_(0xff);
+    }
+    delay(2);
+    this->command_(CMD_DISPLAY_REFRESH);
+    this->wait_until_idle_();
+  } else if (mode == CLEAR_MODE_PARTIAL) {
+    this->command_(CMD_DTM1_DATA_START_TRANS);
+    for (int count = 0; count < pixsize; count++) {
+      this->data_(lastbuff_[count]);
+    }
+    delay(2);
+    this->command_(CMD_DTM2_DATA_START_TRANS2);
+    for (int count = 0; count < pixsize; count++) {
+      this->data_(0xff);
+    }
+    delay(2);
+    this->command_(CMD_DISPLAY_REFRESH);
+    this->wait_until_idle_();
+  }
+
+  return 0;
+}
+
+int GDEW0154M09::write_init_list_(const unsigned char *list) {
+  int list_limit = list[0];
+  unsigned char *start_ptr = ((unsigned char *) list + 1);
+  for (int i = 0; i < list_limit; i++) {
+    this->command_(*(start_ptr + 0));
+    for (int dnum = 0; dnum < *(start_ptr + 1); dnum++) {
+      this->data_(*(start_ptr + 2 + dnum));
+    }
+    start_ptr += (*(start_ptr + 1) + 2);
+  }
+  return 0;
+}
+
+size_t GDEW0154M09::get_buffer_length_() {
+  return size_t(this->get_width_internal()) * size_t(this->get_height_internal()) / 8u;
+}
+
+void HOT GDEW0154M09::display_() {
+  // this->clear_dsram_();
+  this->set_draw_addr_(0, 0, GDEW0154M09_WIDTH, GDEW0154M09_HEIGHT);
+  this->draw_buff_(this->lastbuff_, this->buffer_, this->get_buffer_length_());
+}
+
+void GDEW0154M09::draw_buff_(uint8_t *lastbuff, uint8_t *buff, size_t size) {
+  command_(CMD_DTM1_DATA_START_TRANS);
```

Comment:
This is just one random line, please change to consistently use `this->`
```suggestion
  this->command_(CMD_DTM1_DATA_START_TRANS);
```